src/image/loading_displaying.js

```diff
@@ -539,6 +539,17 @@ p5.prototype.image = function(
 
   let vals = { x: _dx, y: _dy, h: _dh, w: _dw };
   if (fit) {
+    switch (this._renderer._imageMode) {
```

Comment:
This is close to what we want! I think the image mode would change the destination box size (and shift the other way for `CENTER`, i.e.:
```js
switch (this._renderer._imageMode) {
  case constants.CENTER:
    _dx -= _dw * 0.5;
    _dy -= _dh * 0.5;
    break;
  case constants.CORNERS:
    _dw -= _dx;
    _dh -= _dy;
    break;
}
```

...although I wonder if we can reduce the complexity by using `modeAdjust`, since that does these adjustments already? Something like this, where you pass `vals.w`, `vals.h`, `vals.x` and `vals.y` for `dw`, `dh`, dx`, and `dy`:

```js
const vals = canvas.modeAdjust(_dx, _dy, _dw, _dh, this._renderer._imageMode);

if (fit) {
  if (fit === constants.COVER) {
    const { x, y, w, h } = _imageCover(
        xAlign,
        yAlign,
        vals.w, // Replacing dw
        vals.h, // Replacing dh
        _sx,
        _sy,
        _sw,
        _sh
      );
      // etc
```

Reply:
I tried using the existing `modeAdjust` function but the way it works is by shifting the `dx or dy` so as the image appears to be drawn from the center. The reason I couldn't use it to is because it in fitMode `COVER` moving away the destination box makes the image not cover the whole required box, which seems a bit counter intuitive.

Reply:
I think the complexity can be addressed by taking this logic outside into one function call.

Reply:
> in fitMode COVER moving away the destination box makes the image not cover the whole required box

I think `sx` and `sy`, `sw`, and `sh` only change what part of the source image gets put into the destination box, wherever the destination box is, so when you shift the destination box, the source image shifts with it. This would be consistent with how `imageMode` works with `image` when you don't provide any fit parameters: changing from `imageMode(CORNER)` to `imageMode(CENTER)` just shifts where the image is drawn, but the image itself is the same as before.

Let me know if I'm misinterpreting what your concern is, but I think your `_imageCover` and `_imageContain` functions will produce the expected output if we give them updated `dx`, `dy`, `dw`, and `dh` values, either from manually applying shifts as we're doing currently, or by giving them the outputs of `modeAdjust`.